modules/unsupported/flatgeobuf/src/test/java/org/geotools/data/flatgeobuf/FlatGeobufDataStoreTest.java

```diff
@@ -863,6 +865,54 @@ public void readCountriesBbox() throws IOException {
         }
     }
 
+    @Test
+    public void readCountriesFids() throws IOException {
+        SimpleFeatureSource featureSource = getFeatureSource("countries");
+        SimpleFeatureType schema = featureSource.getSchema();
+        FilterFactory2 ff = CommonFactoryFinder.getFilterFactory2();
+        Filter filter = ff.id(ff.featureId("countries.3"), ff.featureId("countries.45"));
+        Query query = new Query(schema.getTypeName(), filter);
+        SimpleFeatureCollection featureCollection = featureSource.getFeatures(query);
+        try (SimpleFeatureIterator it = featureCollection.features()) {
+            int count = 0;
+            while (it.hasNext()) {
+                it.next();
+                count++;
+            }
+            assertEquals(2, count);
+        }
+    }
+
+    @Test(expected = NumberFormatException.class)
+    public void readCountriesFidsInvalid() throws IOException {
+        SimpleFeatureSource featureSource = getFeatureSource("countries");
+        SimpleFeatureType schema = featureSource.getSchema();
+        FilterFactory2 ff = CommonFactoryFinder.getFilterFactory2();
+        Filter filter = ff.id(ff.featureId("countries.noreallong"));
+        Query query = new Query(schema.getTypeName(), filter);
+        SimpleFeatureCollection featureCollection = featureSource.getFeatures(query);
+        try (SimpleFeatureIterator it = featureCollection.features()) {
+            while (it.hasNext()) {
+                it.next();
+            }
+        }
+    }
+
+    @Test(expected = NoSuchElementException.class)
```

Comment:
Same here, it.hasNext should just return false and you get nothing back, including no exception.